### ISSUE TITLE
no_log always applies to with_items

### ISSUE THREAD

**Issue by evanccnyc** [ansible/ansible#12611]
Ansbile Version:

```
ansible 2.0.0 (devel 6c190aa8a4) last updated 2015/10/01 10:18:59 (GMT -400)
  lib/ansible/modules/core: (detached HEAD a4ac171a07) last updated 2015/09/30 12:41:48 (GMT -400)
  lib/ansible/modules/extras: (detached HEAD 3c8c716801) last updated 2015/09/30 12:41:50 (GMT -400)
  config file = /Users/ecarter/Dropbox (SeatGeek)/ansible-seatgeek/ansible.cfg
  configured module search path = None
```

OS X 10.11

Playbook:

```
- hosts: localhost
  connection: local
  gather_facts: False
  tasks:
  - name: configure sudoers
    shell: "echo hello world"
    no_log: True
```

Running with -v in this example but it shows the same problems when just running it without -v and you have multiple items.

Expected Results (works in 1.9):

```
PLAY [localhost] **************************************************************

TASK: [configure sudoers] *****************************************************
changed: [localhost] => {"censored": "results hidden due to no_log parameter", "changed": true, "rc": 0}

PLAY RECAP ********************************************************************
localhost                  : ok=1    changed=1    unreachable=0    failed=0

```

Actual results 2.0 dev branch:

```
Using /Users/ecarter/Dropbox (SeatGeek)/ansible-seatgeek/ansible.cfg as config file

PLAY ***************************************************************************

TASK [configure sudoers] *******************************************************
changed: [localhost] => {"changed": true, "cmd": "echo hello world", "delta": "0:00:00.004131", "end": "2015-10-02 11:06:45.695044", "rc": 0, "start": "2015-10-02 11:06:45.690913", "stderr": "", "stdout": "hello world", "stdout_lines": ["hello world"], "warnings": []}

PLAY RECAP *********************************************************************
localhost                  : ok=1    changed=1    unreachable=0    failed=0
```

**Comment by bcoca** [comment]
I was just updating this last night, try refreshing devel, it seems to work for me with your test play:

```
#>ansible-playbook play.yml  -v
Using /etc/ansible/ansible.cfg as config file
 ______ 
< PLAY >
 ------ 
        \   ^__^
         \  (oo)\_______
            (__)\       )\/\
                ||----w |
                ||     ||


 __________________________ 
< TASK [configure sudoers] >
 -------------------------- 
        \   ^__^
         \  (oo)\_______
            (__)\       )\/\
                ||----w |
                ||     ||


changed: [localhost] => {"censored": "the output has been hidden due to the fact that 'no_log: true' was specified for this result"}
 ____________ 
< PLAY RECAP >
 ------------ 
        \   ^__^
         \  (oo)\_______
            (__)\       )\/\
                ||----w |
                ||     ||


localhost                  : ok=1    changed=1    unreachable=0    failed=0   
```

**Comment by evanccnyc** [comment]
I now see no_log applying everywhere not on a per task basis.

@bcoca

**Comment by bcoca** [comment]
added extra tag to play

```
- hosts: localhost
  connection: local
  gather_facts: False
  tasks:
  - name: configure sudoers
    shell: "echo hello world"
    no_log: True
  - name: configure sudoers
    shell: "echo hello world"
```

here are the results:

``````
#> ansible-playbook play.yml  -v
Using /etc/ansible/ansible.cfg as config file
 ______ 
< PLAY >
 ------ 
          \      (__)      
           \     /oo|  
            \   (_"_)*+++++++++*
                   //I#\\\\\\\\I\
                   I[I|I|||||I I `
                   I`I'///'' I I
                   I I       I I
                   ~ ~       ~ ~
                     Scowleton


 __________________________ 
< TASK [configure sudoers] >
 -------------------------- 
          \      (__)      
           \     /oo|  
            \   (_"_)*+++++++++*
                   //I#\\\\\\\\I\
                   I[I|I|||||I I `
                   I`I'///'' I I
                   I I       I I
                   ~ ~       ~ ~
                     Scowleton


changed: [localhost] => {"censored": "the output has been hidden due to the fact that 'no_log: true' was specified for this result"}
 __________________________ 
< TASK [configure sudoers] >
 -------------------------- 
          \      (__)      
           \     /oo|  
            \   (_"_)*+++++++++*
                   //I#\\\\\\\\I\
                   I[I|I|||||I I `
                   I`I'///'' I I
                   I I       I I
                   ~ ~       ~ ~
                     Scowleton


changed: [localhost] => {"changed": true, "cmd": "echo hello world", "delta": "0:00:00.002885", "end": "2015-10-02 11:25:11.929623", "rc": 0, "start": "2015-10-02 11:25:11.926738", "stderr": "", "stdout": "hello world", "stdout_lines": ["hello world"], "warnings": []}
 ____________ 
< PLAY RECAP >
 ------------ 
          \      (__)      
           \     /oo|  
            \   (_"_)*+++++++++*
                   //I#\\\\\\\\I\
                   I[I|I|||||I I `
                   I`I'///'' I I
                   I I       I I
                   ~ ~       ~ ~
                     Scowleton


localhost                  : ok=2    changed=2    unreachable=0    failed=0   ```
``````

**Comment by evanccnyc** [comment]
@bcoca 
Try it with this:

```
- hosts: localhost
  connection: local
  gather_facts: False
  tasks:
  - name: configure sudoers
    shell: "echo hello world"
    no_log: True
  - name: configure sudoers
    shell: "echo hello world"
    with_random_choice:
     - "go through the door"
     - "drink from the goblet"
     - "press the red button"
     - "do nothing"
```

**Comment by evanccnyc** [comment]
It appears to occur on any of the  with_\* ones.

**Comment by bcoca** [comment]
confirmed, with items always applies no_log (probably i broke something with last patch)

**Comment by bcoca** [comment]
fixed  via e2ae3215f6e8baaa1a01dad590b8a070940190ce, also added test cases for both no_log being true or false so we don't censor all by mistake again
